Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveySettings.kt

```diff
@@ -1,7 +1,27 @@
 package de.rki.coronawarnapp.datadonation.survey
 
+import android.content.Context
+import de.rki.coronawarnapp.datadonation.OneTimePassword
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.clearAndNotify
+import de.rki.coronawarnapp.util.preferences.createFlowPreference
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class SurveySettings @Inject constructor()
+class SurveySettings @Inject constructor(
+    @AppContext val context: Context
+) {
+
+    private val preferences by lazy {
+        context.getSharedPreferences("survey_localdata", Context.MODE_PRIVATE)
+    }
+
+    val oneTimePassword =
+        preferences.createFlowPreference<OneTimePassword?>(
```

Comment:
I don't think this works as `OneTimePassword` is not a primitive type.
This will probably crash as there will be no default mapper for this in `FlowPreference.kt`.

If you want to save the whole data class you can use a gson writer, see `gsonReader` and `gsonWriter`.